conf.py

```diff
@@ -103,3 +104,96 @@
                                   "README.html")): # don't create dangling symlinks;
                                                    # it annoyes Github Pages.
                 os.symlink("README.html", z)
+
+
+
+
+# -- Custom scripts ----------------------------------------------------------
+
+import os
+from pathlib import Path
+import random
+import requests
+from subprocess import run
+from textwrap import dedent
+from urllib.parse import urlparse
+from ghapi.all import GhApi
+import pandas as pd
+
+import yaml
+
+from sphinx.application import Sphinx
+from sphinx.util import logging
+
+LOGGER = logging.getLogger("conf")
+
+def build_gallery(app: Sphinx):
```

Comment:
It's inelegant to do all this in the conf.py file, running it on *every* parse like this.

It would be significantly cleaner to package this script as a [custom directive](https://myst-parser.readthedocs.io/en/latest/syntax/optional.html#optional-myst-syntaxes), stick it up on pypi, and then instead of 

````
```{include} gallery.txt

```
````

do

````
```{gallery}
@@ -0,0 +1,20 @@
- name: Spinal Cord Toolbox
  website: https://spinalcordtoolbox.com/
  repository: https://github.com/spinalcordtoolbox/spinalcordtoolbox
  image: https://raw.githubusercontent.com/spinalcordtoolbox/spinalcordtoolbox/master/documentation/source/_static/img/logo_sct.png
- name: ivadomed
  website: https://ivadomed.org
  repository: https://github.com/ivadomed/ivadomed
  image: https://raw.githubusercontent.com/ivadomed/ivadomed/master/images/ivadomed_logo.png
- name: qMRLab
  website: https://qmrlab.readthedocs.io
  repository: https://github.com/qMRLab/qMRLab
  image: https://raw.githubusercontent.com/qMRLab/documentation/master/logo/qMR_logo_orig.png
- name: Shimming Toolbox
  website: https://shimming-toolbox.org
  repository: https://github.com/shimming-toolbox
  image: https://raw.githubusercontent.com/shimming-toolbox/shimming-toolbox/master/docs/source/_static/img/shimming_toolbox_logo.png
- name: AxonDeepSeg
  website: https://axondeepseg.readthedocs.io/
  repository: https://github.com/neuropoly/axondeepseg
  image: https://raw.githubusercontent.com/neuropoly/axondeepseg/master/docs/source/_static/logo_ads-alpha.png
```
````

Reply:
and those images should be inlined

Reply:
agreed-- but i don't have the time to do it, so https://github.com/neuropoly/neuro.polymtl.ca/issues/47